### DISCUSSION TITLE
[4.0] Trigger an event on article/module/menu when order change by drag and drop

### DISCUSSION THREAD

**Discussion by DevMatGod** [joomla/joomla-cms#47181] [2. Cold Features]
### Issue
There is no way to create a plugin that triggers an event when the order of an item (Article, Module or Menu) is changed by dragging it in the list.

### Solution
Trigger onContentAfterSave or something similar when the order is changed via drag-and-drop.

**Comment by chetanmadaan** [comment]
What use would this serve?

**Comment by mbabker** [comment]
> What use would this serve?

Aside from the fact that most every write action should have some kind of pre- and post- hook?  Why should some write actions have extension points that can be hooked to do extra things and others not?

**Comment by chetanmadaan** [comment]
Thanks Michael.

**Comment by DevMatGod** [comment]
In my case, I need to refresh the cache every time an administrator change something.  But it could be useful to someone who needs to pre-render something relative to the order. Right now, he'll need to trust that the user would open and save the article.

When you change the order, the change is saved in the database. I think it should be treated as such

**Comment by alikon** [comment]
Closing this due to not receiving required information to determine if this is a bug or not. If you feel this still needs review, please open a new tracker entry with as much information as possible to ensure it can be reviewed properly

**Comment by mbabker** [comment]
All of the required information is there!

This is a feature request to trigger before/after events on drag and drop reordering.  Hell, it could easily be done if the controller actions handling that were using the existing model `save()` method which triggers the `onContentBeforeSave` and `onContentAfterSave` events.

If you're closing feature requests because nobody has acted upon them, then use a different copy/paste blob that's more appropriate than "oh this doesn't have all the info we want, closing".
